File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.kt
package lv0.코딩기초트레이닝.연산과조건문.홀짝에따라다른값반환하기
class Solution {
    fun solution(n: Int): Int {
        var answer: Int = 0
        val isEven = n % 2 == 0
        val start = if (isEven) 2 else 1
        for (i in start..n step 2) {
            answer += if (isEven) i * i else i
        }
        return answer
    }
}

File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.go
package main

func solution(n int) int {
	answer := 0
	isEven := n%2 == 0
	start := 0

	if isEven {
		start = 2
	} else {
		start = 1
	}
	for i := start; i <= n; i += 2 {
		if isEven {
			answer += i * i
		} else {
			answer += i
		}
	}

	return answer
}
